### PR TITLE
DFR-2645:  INC5581298 - send order to deal with approved order withou…

### PR DESCRIPTION
**Pull request by rishikrsharma**
### Jira link (if applicable)
DFR-2645

### Change description ###
INC5581298 - send order to deal with approved order without document

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/sendorder/SendOrderPartyDocumentHandler.java [hunk 1]
```diff
@@ -122,13 +122,17 @@ private void addAdditionalOrderDocumentToPartyCollection(FinremCaseData caseData
...
-                .orElse(new ArrayList<>());
+            .orElse(new ArrayList<>());
         if (existingCollection.isEmpty()) {
             return true;
         }
-        return existingCollection.stream().noneMatch(doc -> doc.getValue().getApproveOrders().stream().anyMatch(order ->
-                order.getValue().getCaseDocument().getDocumentFilename().equals(ADDITIONAL_HEARING_FILE_NAME)
-                        && order.getValue().getCaseDocument().getDocumentUrl().equals(document.getDocumentUrl())
+        return existingCollection.stream().noneMatch(doc -> doc.getValue().getApproveOrders().stream().anyMatch(order -> {
+                if (order.getValue().getCaseDocument() != null) {
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/sendorder/SendOrderPartyDocumentHandler.java:130]: Could you extract this if clause to a method to make it a bit cleaner to follow?

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/sendorder/SendOrderPartyDocumentHandler.java [hunk 2]
```diff
@@ -122,13 +122,17 @@ private void addAdditionalOrderDocumentToPartyCollection(FinremCaseData caseData
...
+                    return order.getValue().getCaseDocument().getDocumentFilename().equals(ADDITIONAL_HEARING_FILE_NAME)
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/sendorder/SendOrderPartyDocumentHandler.java:131]: maybe a method for this condition too to make it clearer what this is returning.

**Review by gmikerlemos — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/sendorder/SendOrderPartyDocumentHandler.java (hunk 1 above)
Comment (rishikrsharma) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/sendorder/SendOrderPartyDocumentHandler.java]: made change

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/sendorder/SendOrderPartyDocumentHandler.java (hunk 2 above)
Comment (rishikrsharma) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/sendorder/SendOrderPartyDocumentHandler.java]: made change

**Review by srochani — APPROVED** [review]

**Review by gmikerlemos — APPROVED** [review]

**Pull request merged**